src/components/Identity/createComponent.js

```diff
@@ -44,16 +44,14 @@ export default ({
         }
       },
       getIdentity: {
-        optionsValidator: options => {
-          return getIdentityOptionsValidator(options);
-        },
+        optionsValidator: getIdentityOptionsValidator,
         run: options => {
           return consent.awaitConsent().then(() => {
             if (ecid) {
-              return ecid;
+              return { ecid };
             }
             return getIdentity(options.namespaces).then(() => {
-              return ecid;
+              return { ecid };
```

Comment:
Ooh. Yes. I'm glad you caught this.

Reply:
I believe the key here should match the namespace being passed in. In other words, if the validator states that the user must pass in `ECID`, then they key we should be returning here should be `ECID` and not `ecid`.

Reply:
@jileeshs I see this as a blocker and will therefore request changes, but other than that I think everything else looks good. I would like to have @jonsnyder take a look at your validator, but I wouldn't let it hold up merging.

Reply:
@jileeshs and I talked over slack to come up with the validator.